field/composite_test.go

```diff
@@ -1892,3 +1894,60 @@ func TestComposite_getFieldIndexOrTag(t *testing.T) {
 		require.Empty(t, index)
 	})
 }
+
+func TestComposit_concurrency(t *testing.T) {
+	t.Run("Pack and Marshal", func(t *testing.T) {
+		// packing and marshaling
+		data := &TLVTestData{
+			F9A:   NewHexValue("210720"),
+			F9F02: NewHexValue("000000000501"),
+		}
+
+		composite := NewComposite(tlvTestSpec)
+
+		wg := sync.WaitGroup{}
+		wg.Add(5)
```

Comment:
Nit: Do this inside of the go routine before the defer and change it to `wg.Add(1)` that way the for loop is the arbiter of modifying the waitgroup.